### PR TITLE
support loadSubpackage for taobao mini game

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by bofeng-song** [comment]
https://github.com/cocos/3d-tasks/issues/15601

File: platforms/minigame/common/engine/AssetManager.js [hunk 1]
```diff
@@ -210,14 +210,23 @@ function downloadBundle (nameOrUrl, options, onComplete) {
     const suffix = version ? `${version}.` : '';
 
     if (subpackages[bundleName]) {
-        var config = `subpackages/${bundleName}/config.${suffix}json`;
+        var config;
+        if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
+            config = `${bundleName}/config.${suffix}json`;
+        } else {
+            config = `subpackages/${bundleName}/config.${suffix}json`;
+        }
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js:226]: Why taobao mini game has different pash?

File: platforms/minigame/common/engine/AssetManager.js [hunk 2]
```diff
@@ -210,14 +210,23 @@ function downloadBundle (nameOrUrl, options, onComplete) {
...
                 onComplete(err, null);
                 return;
             }
             downloadJson(config, options, (err, data) => {
-                data && (data.base = `subpackages/${bundleName}/`);
+                if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
+                    data && (data.base = `${bundleName}/`);
+                } else {
+                    data && (data.base = `subpackages/${bundleName}/`);
+                }
```
Comment (minggo) [inline platforms/minigame/common/engine/AssetManager.js]: The same question as above.

File: platforms/minigame/platforms/taobao-mini-game/wrapper/fs-utils.js [hunk 3]
```diff
@@ -244,9 +244,8 @@ var fsUtils = {
                 onComplete && onComplete(new Error(`Failed to load subpackage ${name}: ${res.errorMessage}`));
             },
         });
-        task = task || {};
         // onProgress && task.onProgressUpdate(onProgress);
-        return task;
+        return {};
```
Comment (PPpro) [inline platforms/minigame/platforms/taobao-mini-game/wrapper/fs-utils.js:248]: why don't return task object

File: platforms/minigame/common/engine/AssetManager.js (hunk 1 above)
Comment (bofeng-song) [reply platforms/minigame/common/engine/AssetManager.js:226]: Platform Interface Restrictions. taobao creative app does not support subpackage

File: platforms/minigame/platforms/taobao-mini-game/wrapper/fs-utils.js (hunk 3 above)
Comment (bofeng-song) [reply platforms/minigame/platforms/taobao-mini-game/wrapper/fs-utils.js:248]: The return value type of loadSubpackage is void

**Review by PPpro — APPROVED** [review]

File: platforms/minigame/common/engine/AssetManager.js (hunk 1 above)
Comment (minggo) [reply platforms/minigame/common/engine/AssetManager.js:226]: It is better to wrap it into a function, for example, getConfigPath.

File: platforms/minigame/common/engine/AssetManager.js [hunk 4]
```diff
@@ -209,15 +209,32 @@ function downloadBundle (nameOrUrl, options, onComplete) {
     const version = options.version || cc.assetManager.downloader.bundleVers[bundleName];
     const suffix = version ? `${version}.` : '';
 
+    function getConfigPathForSubPackage () {
+        if (sys.platform === sys.Platform.TAOBAO_MINI_GAME) {
+            return `${bundleName}/config.${suffix}json`;
+        }
+        return `subpackages/${bundleName}/config.${suffix}json`;
```
Comment (dumganhar) [inline platforms/minigame/common/engine/AssetManager.js:216]: Why `taobao minigame` doesn't need the `subpackages` prefix?

Comment (bofeng-song) [reply platforms/minigame/common/engine/AssetManager.js:216]: taobao mini game depends on config of game.json, subpackages has been config.
And taobao's filesystem does not support other directory before bundle name

Comment (bofeng-song) [reply platforms/minigame/common/engine/AssetManager.js:216]: loadsubpackage's example for taobao.
image

**Review by dumganhar — APPROVED** [review]

**Pull request merged**